src/com/company/zadania/obiektowe/zadanie1/Zamowienie.java

```diff
@@ -16,28 +18,64 @@ public Zamowienie(int maksRozmiar) {
     private int ileDodanych = 0;
 
     public void dodajPozycje(Pozycja p) {
+        for (Pozycja value : pozycja) {
+            if (value == null) continue;
+            if (p.getNazwaTowaru().equals(value.getNazwaTowaru())) {
+                value.setIleSztuk(p.getIleSztuk());
+                return;
+            }
+        }
         pozycja[ileDodanych] = p;
         ileDodanych++;
     }
 
     public double obliczWartosc() {
         double wartoscZamowienia = 0;
-        for (int i = 0; i < pozycja.length; i++) {
-            if(pozycja[i]==null) break;
-            wartoscZamowienia += pozycja[i].obliczWartosc();
+        for (Pozycja value : pozycja) {
+            if (value == null) continue;
+            wartoscZamowienia += value.obliczWartosc();
         }
         return wartoscZamowienia;
     }
 
+    public void usunPozycje(int indeks) {
+        try {
+            pozycja[indeks] = null;
+        } catch (ArrayIndexOutOfBoundsException e) {
```

Comment:
jak nie chcesz wyjatkow to powinno byc 
public boolean usunPozycje (int index) { 
  try { poz[] = null,; return true; } catch { log.err  return false; }